internals/overlord/logstate/loki/loki.go

```diff
@@ -88,9 +89,7 @@ func (c *Client) SetLabels(serviceName string, labels map[string]string) {
 
 	// Make a copy to avoid altering the original map
 	newLabels := make(map[string]string, len(labels)+1)
-	for k, v := range labels {
-		newLabels[k] = v
-	}
+	maps.Copy(newLabels, labels)
```

Comment:
I reckon this would be more idiomatic:

```suggestion
    newLabels := maps.Clone(labels)
```